rcode-reader/src/main/java/eu/rssw/pct/RCodeInfo.java

```diff
@@ -0,0 +1,723 @@
+import eu.rssw.pct.elements.PropertyElement;
+import eu.rssw.pct.elements.QueryElement;
+import eu.rssw.pct.elements.TableElement;
+import eu.rssw.pct.elements.VariableElement;
+
+/**
+ * Import debug segment information from rcode.
+ */
+public class RCodeInfo {
+  // Magic number, followed by same magic number written as little-endian
+  private static final int MAGIC1 = 0x56CED309;
+  private static final int MAGIC2 = 0x09D3CE56;
+
+  // Header values
+  private static final int HEADER_SIZE = 68;
+  private static final int HEADER_OFFSET_MAGIC = 0;
+  private static final int HEADER_OFFSET_TIMESTAMP = 4;
+  private static final int HEADER_OFFSET_MD5 = 10;
+  private static final int HEADER_OFFSET_RCODE_VERSION = 14;
+  private static final int HEADER_OFFSET_SEGMENT_TABLE_SIZE = 0x1E;
+  private static final int HEADER_OFFSET_SIGNATURE_SIZE = 56;
+  private static final int HEADER_OFFSET_TYPEBLOCK_SIZE = 60;
+  private static final int HEADER_OFFSET_RCODE_SIZE = 64;
+
+  // Segment table values
+  private static final int SEGMENT_TABLE_OFFSET_INITIAL_VALUE_SEGMENT_OFFSET = 0;
+  private static final int SEGMENT_TABLE_OFFSET_INITIAL_VALUE_SEGMENT_SIZE = 16;
+  private static final int SEGMENT_TABLE_OFFSET_DEBUG_SEGMENT_OFFSET = 12;
+  private static final int SEGMENT_TABLE_OFFSET_DEBUG_SEGMENT_SIZE = 28;
+
+  private ByteOrder order;
+  private int version;
+  private boolean sixtyFourBits;
+  private long timeStamp;
+  private int md5;
+
+  private int segmentTableSize;
+  private int signatureSize;
+  private int typeBlockSize;
+  private int rcodeSize;
+  private int initialValueSegmentOffset;
+  private int initialValueSegmentSize;
+  private int debugSegmentOffset;
+  private int debugSegmentSize;
+
+  // From type block
+  private boolean isClass = false;
+  private String typeName;
+  private String parentTypeName;
+  private String assemblyName;
+  private List<String> interfaces = new ArrayList<>();
+
+  private RCodeUnit unit = new RCodeUnit();
+
+  public RCodeInfo(InputStream input) throws InvalidRCodeException, IOException {
+    this(input, null);
+  }
+
+  /**
+   * Parse InputStream and store debug segment information
+   * 
+   * @param input Has to be closed by caller
+   * @param out Output stream for debug. Can be null
+   * 
+   * @throws InvalidRCodeException
+   * @throws IOException
+   */
+  public RCodeInfo(InputStream input, PrintStream out) throws InvalidRCodeException, IOException {
+    processHeader(input, out);
+    processSignatureBlock(input, out);
+    processSegmentTable(input, out);
+
+    if ((debugSegmentOffset > 0) && (debugSegmentSize > 0)) {
+      long bytesRead = input.skip(debugSegmentOffset);
+      if (bytesRead != debugSegmentOffset) {
+        throw new InvalidRCodeException("Not enough bytes to reach debug segment");
+      }
+      processDebugSegment(input, out);
+    }
+
+    if (typeBlockSize > 0) {
+      long bytesRead = input.skip(rcodeSize - debugSegmentOffset - debugSegmentSize);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this subtraction operation to a "long". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2184)